genomic_data_service/region_indexer_local_file_reader.py

```diff
@@ -0,0 +1,73 @@
+from asyncio.log import logger
+import csv
+import gzip
+
+
+class LocalSnpReader():
+    def __init__(self, file_path):      
+        self.file_path = file_path
+        self.file = gzip.open(self.file_path, mode='rt')
+
+
+    def parse(self):
+
+        reader = csv.reader(self.file, delimiter='\t')
+        for row in reader:
+            if row[0].startswith('#'):
+                continue
+
+            try:
+                (chrom, doc) = self.snp(row)
+           
+            except Exception:
+                logger.error('%s - failure to parse row %s:%s:%s, skipping row',
+                          self.file_path, row[0], row[1], row[2])
+                continue
+
+            yield (chrom, doc)
+
+
+    def snp(self, row):
```

Comment:
Maybe we could avoid duplicating this code by making a base class?

Reply:
The reason I separated them for now is that I feel S3BedFileRemoteReader also needs some improvement itself. S3BedFileRemoteReader bring in the whole file_properties as its parameter but in fact it only need file_size and s3_uri. I wanted to change its constructor if we want to modify the file reader. I know we discussed to write a base class previously. I just don't want to do it in this ticket so we can talk about it before I start to mess it around.